### PR TITLE
Sewerwater additional fixes

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard1**
## Summary

The individual sewer installation data was incorrectly shown in the tooltips due to a misunderstanding. This is fixed now.
Plus a dashed line is added for any missing averages. (most of the time the last week is missing).
Also the description in the averages tile is fixed.

## Motivation

Feature request

## Detailed design

The scatter data and average data is first compared to see if there are any date differences. If so, an extra series is calculated for the remaining days at the end of the averages. This is to visualize the missing data.

## Drawbacks

n/a

## Alternatives

n/a

## Unresolved questions

n/a

### Governance

- [ ] Documentation is added
- [ ] Test cases are added or updated
- [ ] I've read the contributing document https://github.com/minvws/.github/blob/master/CONTRIBUTING.md
- [ ] I've read and understand the Code of Conduct https://github.com/minvws/.github/blob/master/CODE_OF_CONDUCT.md
- [ ] I understand that any contributions or suggestions I made may make it into the actual code. I've read the License
      https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/LICENSE.txt
      and the contributor license agreement https://github.com/minvws/nl-covid19-notification-app-coordination/blob/master/CLA.md

### CODE REVIEW

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts [hunk 1]
```diff
@@ -13,6 +13,35 @@ type Week = {
   end: number;
 };
 
+function calculateDaysBetween(date1: number, date2: number) {
+  const ONE_DAY = 1000 * 60 * 60 * 24;
+
+  const date1_ms = date1 * 1000;
```
Comment (VWSCoronaDashboard8) [inline src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:19]: Are we programming in Rust or C now? 😄 Or did I miss the memo that we are moving to snake-cased variables?

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts [hunk 2]
```diff
@@ -13,6 +13,35 @@ type Week = {
...
+  return Math.floor(difference_ms / ONE_DAY);
+}
+
+/**
+ * This function calculates the difference in days between the last RWZI measurement
+ * and the last average and returns an array of data that is used to draw the dashed line
+ * at the end of the averages.
+ */
+function createRemainingDaysData(value: Value, maxDate: number) {
+  const datapointLength =
```
Comment (VWSCoronaDashboard8) [inline src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:33]:
```suggestion
  const dataPointLength =
```

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts [hunk 3]
```diff
@@ -13,6 +13,35 @@ type Week = {
...
+    calculateDaysBetween(value.week_start_unix, maxDate) + 1;
+
+  const data = [...new Array(datapointLength)];
```
Comment (VWSCoronaDashboard8) [inline src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:36]: If you are already creating a new array, what's the point of destructuring and creating yet another new array? This is creating a new array twice.

Also, you are passing a length to the array constructor like it is C or something, but in js arrays are always dynamic length, right? So the wording makes it confusing for me.

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts [hunk 4]
```diff
@@ -44,19 +73,28 @@ export function useRegionalSewerWaterChartOptions(
       end: value.week_end_unix,
     }));
 
-    const weekSets: Record<string, Week[]> = {
+    let maxDate = filteredAverageValues.reduce((max, value) => {
+      return Math.max(max, value.week_end_unix);
+    }, 0);
+    maxDate = filteredScatterPlotValues.reduce((max, value) => {
```
Comment (VWSCoronaDashboard8) [inline src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:79]: This looks overly complicated. You could do one reduce and return the date_measurement_unix or the week_end_unix value depending on which one is higher.

**Review by VWSCoronaDashboard8 — APPROVED** [review]

**Review by VWSCoronaDashboard4 — APPROVED** [review]

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts (hunk 3 above)
Comment (VWSCoronaDashboard1) [reply src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:36]: I'm creating an array with an initial length. That way I can subsequently populate the array with the map() function. There's a weird issue (bug?) in JS where if you create an array with a fixed length the indexes aren't generated. Therefore we use the trick [...new Array(length)].
This is quite widely used, I'm nit sure what the problem is...

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts (hunk 4 above)
Comment (VWSCoronaDashboard1) [reply src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:79]: One reduce how? These are two separate arrays. filteredAverageValues and filteredScatterPlotValues.

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts (hunk 1 above)
Comment (VWSCoronaDashboard1) [reply src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts]: Fixed

File: src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts (hunk 4 above)
Comment (VWSCoronaDashboard8) [reply src/components/lineChart/hooks/useRegionalSewerWaterChartOptions.ts:79]: Forgive me. I didn't read it properly.

**Pull request merged**